accepted/0000-hub-online-synchronization.md

```diff
@@ -0,0 +1,228 @@
+
+The new Hub synchronization would be focused on HUB online deployments to allow scale SUSE Manager infrastructure. The only focus of this solution would be to work in scale and make it easier to use.
+
+
+# Detailed design
+[design]: #detailed-design
+
+The solution focuses on re-using the repo-sync mechanism to synchronize channels from Hub server to peripheral servers. It will be described in several steps that aim to address how it will be used and how we can technically support it.
+
+The main focus is to develop the integration between HUB server and peripherals through API only and re-using existing API methods as much as possible. However, some new custom methods will be needed.
+
+Configuration should be pushed from the HUB to peripherals, to avoid the need of direct management of the peripherals.
+
+For now this solution is focused on software channels only, but can be extended to synchronize more data types.
+
+
+## Define connection between HUB and Peripherals
+
+We can follow a similar approach to what exists on ISSv1. On the hub side we can define multiple peripheral servers to connect to by providing the FQDN and an authentication token.
+On the peripheral side we also need to define the Hub server FQDN and an Authentication token.
+
+Each peripheral server can only have 1 Hub server (master). This will avoid dealing with problems like channel label conflicts between multiple masters. Configure a hub server will block access to a set of menu items like: "Admin" -> "Setup wizard" -> "Organization Credentials" and "Products" and "PAYG Connection" (similar to what we already have for ISSv1).
+We cannot add Hub connection if SCC credentials are defined, they should be mutually excluded.
+
+We can re-use and improve existing ISSv1 database tables to save the needed data.
+
+
+## Hub as a proxy for SCC data
+
+The SUSE Manager server needs a set of metadata to be able to operate. Currently that metadata is provided by SCC directly or, in the case of PAYG, provided by the cloud RMT infrastructure. We should also provide this data in SUSE Manager HUB to be consumed by the peripheral servers.
+The minimal endpoints to be provided are:
+   - "/suma/product_tree.json"
+   - "/connect/organizations/products/unscoped"
+
+On top of this, we should also provide an endpoint for peripherals to send status data needed by SCC (example of this is the minions registered and hardware information). Peripheral servers should send this data to the HUB instead of SCC, and the HUB server should consolidate it and send it to SCC.
+
+## Peripheral software channels creation
+
+We need a mechanism to create the channels in the peripheral servers (vendor and CLM's) in the desired organization. The peripheral channel creation must be done automatically from the HUB server through an API. Since we are making special channel creation (defined next), those API methods should be available to server-to-server communication only.
```

Comment:
I think it would be good to define a little bit more in detail, how this server-to-server API should look like.
This should also say something about the exiting API namespaces `sync.master` and `sync.slave`.

- What namespace should be used for it?
- one namespace or multiple? 
- design it with an outlook to the future and what else needs to be added in future to this API. E.g. activation keys, config channels, images, formulas, etc.
- how should the authentication work?

Reply:
Hey Michael. I added some clarification about API namespace and use cases. I didn't add any details about the exact API methods to develop because it looks to me like it's an implementation detail.
Could you have a look if it's more clear now? Thank you